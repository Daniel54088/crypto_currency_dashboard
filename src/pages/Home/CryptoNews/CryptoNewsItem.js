import React from "react";
import PropTypes from "prop-types";

const propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
  desc: PropTypes.string,
  image: PropTypes.string
};
const defaultProps = {
  title: "Default CryptoNews Title",
  subTitle: "Default CryptoNews subTitle",
  desc: "BTC, Crypto, Trading, Tips, Cryptocurrency",
  image: require("../../../assets/news1.jpg")
};

function CryptoNewsItem(props) {
  const { title, subTitle, desc, image } = props;

  return (
    <div className="jr-news-item">
      <div className="jr-news-thumb">
        <img className="rounded-xl" src={image} alt="..." />
      </div>
      <div className="jr-news-content">
        <h4 className="mt-0">{title}</h4>
        <p>{subTitle}</p>
        <div className="jr-news-tags-row">
          <div className="jr-news-tags-left">
            <p className="text-grey text-truncate">
              <i
                className={`zmdi zmdi-label-alt jr-fs-lg mr-2 d-inline-block align-middle`}
              />
              {desc}
            </p>
          </div>
          <div className="jr-news-tags-right">
            <p className="text-primary ml-auto pointer text-truncate">
              Read Full Story 
              <i
                className={`zmdi zmdi-long-arrow-right jr-fs-xxl ml-2 d-inline-block align-middle`}
              />
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default CryptoNewsItem;

CryptoNewsItem.propTypes = propTypes;
CryptoNewsItem.defaultProps = defaultProps;
