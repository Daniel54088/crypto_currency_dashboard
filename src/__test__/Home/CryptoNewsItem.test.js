import React from "react";
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import CryptoNewsItem from '../../pages/Home/CryptoNews/CryptoNewsItem';


it('render correctly CryptoNewsItem component', () => {
    const CryptoNewsItemComponent = renderer.create(<CryptoNewsItem />).toJSON();
    expect(CryptoNewsItemComponent).toMatchSnapshot();
});