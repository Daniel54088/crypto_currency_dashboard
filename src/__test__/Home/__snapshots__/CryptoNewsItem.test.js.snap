// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly CryptoNewsItem component 1`] = `
<div
  className="jr-news-item"
>
  <div
    className="jr-news-thumb"
  >
    <img
      alt="..."
      className="rounded-xl"
      src="test-file-stub"
    />
  </div>
  <div
    className="jr-news-content"
  >
    <h4
      className="mt-0"
    >
      Default CryptoNews Title
    </h4>
    <p>
      Default CryptoNews subTitle
    </p>
    <div
      className="jr-news-tags-row"
    >
      <div
        className="jr-news-tags-left"
      >
        <p
          className="text-grey text-truncate"
        >
          <i
            className="zmdi zmdi-label-alt jr-fs-lg mr-2 d-inline-block align-middle"
          />
          BTC, Crypto, Trading, Tips, Cryptocurrency
        </p>
      </div>
      <div
        className="jr-news-tags-right"
      >
        <p
          className="text-primary ml-auto pointer text-truncate"
        >
          Read Full Story
          <i
            className="zmdi zmdi-long-arrow-right jr-fs-xxl ml-2 d-inline-block align-middle"
          />
        </p>
      </div>
    </div>
  </div>
</div>
`;
